src/main/java/com/firebolt/jdbc/service/FireboltEngineInformationSchemaService.java

```diff
@@ -66,12 +79,16 @@ private Engine getEngine(String engine, @Nullable String database) throws SQLExc
                 if (database != null && !database.equals(attachedDatabase)) {
                     throw new FireboltException(format("The engine with the name %s is not attached to database %s", engine, database));
                 }
-                return new Engine(rs.getString(ENGINE_URL), status, rs.getString(ENGINE_NAME_FIELD), attachedDatabase, null);
+                String[] engineUrl = rs.getString(ENGINE_URL).split("\\?", 2);
+                String engineHost = engineUrl[0];
+                String[] engineQuery = engineUrl.length > 1 ? engineUrl[1].split("&") : new String[0];
+                Arrays.stream(engineQuery).map(prop -> prop.split("=")).filter(a -> a.length == 2).forEach(prop -> properties.addProperty(prop[0], prop[1]));
```

Comment:
Let's pop a comment here what's happening, otherwise a long string of map/reduce is hard to read.